Add bundle progress bar to logs

brentvatne opened the pull request:
It works, see this image. Depends on https://github.com/facebook/react-native/pull/13172

<img width="1094" alt="slack-imgs" src="https://cloud.githubusercontent.com/assets/90494/24517834/99f5bfb0-1534-11e7-841f-c54ae4a0f235.png">

Falls back to this on old SDKs

<img width="700" alt="slack-imgs-1" src="https://cloud.githubusercontent.com/assets/90494/24519240/6409e12e-1539-11e7-856e-7b50f8425cb5.png">

brentvatne commented on react-native-scripts/package.json:
```diff
@@ -33,8 +33,9 @@
+    "progress": "^1.1.8",
     "qrcode-terminal": "^0.11.0",
-    "xdl": "^37.0.0"
+    "xdl": "38.0.0-beta.2"
```
yup!

brentvatne commented on react-native-scripts/src/util/packager.js:
```diff
@@ -52,66 +56,107 @@ function run(onReady: () => ?any, options: Object = {}) {
-          return;
-        }
+    if (packagerReady) {
+      const message = `${chunk.msg.trim()}\n`;
```
``lol``

brentvatne commented on react-native-scripts/src/util/log.js:
```diff
@@ -0,0 +1,35 @@
+
+let _bundleProgressBar;
+
+function respectProgressBars(commitLogs) {
```
@dikaiosune - so just:

```
let _oldConsoleLog = console.log;
console.log = log;

// implement log with _oldConsoleLog
```

dikaiosune replied:
wfm

brentvatne commented on react-native-scripts/package.json:
```diff
@@ -33,8 +33,9 @@
     "match-require": "^2.0.0",
     "minimist": "^1.2.0",
     "path-exists": "^3.0.0",
+    "progress": "^1.1.8",
```
indeed, I don't have a windows machine to test this :<

brentvatne replied:
or rather I do but I have no idea how to use windows for anything except oculus and netflix

brentvatne replied:
fwiw we use this same ProgressBar module on exp so this could be a problem we have to address there too if it is indeed true and not user error in that issue

dikaiosune replied:
ah ok. hm maybe it's broken in exp too then?

brentvatne replied:
can you try out installing exp and init'ing a project? should show progress bar while downloading

dikaiosune replied:
*wipes tear away while spinning up a VM*

dikaiosune replied:
so apparently the inquirer package is also irrevocably broken on git bash, so i'm just going to step away slowly

brentvatne replied:
resolution: don't support git bash for now if you want a pleasant crna experience. powershell recommended on windows for now. if anyone reading this is interested in fixing git bash support it would be welcome!

brentvatne commented:
ok I think I addressed everything. you can test this out by cloning and running `npm pack` in react-native-scripts then running `create-react-native-app FancyApp --scripts-version /path/to/create-react-native-app/react-native-scripts/react-native-scripts-0.0.26.tgz` -- I updated dependencies for each package required to make this work

brentvatne commented on react-native-scripts/template/app.json:
```diff
@@ -1,5 +1,5 @@
 {
   "expo": {
-    "sdkVersion": "15.0.0"
+    "sdkVersion": "UNVERSIONED"
```
of course we will change this before merging, just changed it for now so you can test

brentvatne commented:
we can almost merge this @dikaiosune ;)

dikaiosune commented:
Woooooooo

Is it just waiting on sdk 16?

dikaiosune commented:
Not sure why it is showing changes vs. master that I rebased into the branch, but I'm testing this now.

dikaiosune commented on react-native-scripts/src/scripts/init.js:
```diff
@@ -94,8 +95,8 @@ module.exports = async (appPath: string, appName: string, verbose: boolean, cwd:
-  console.log(`Installing dependencies using ${command}...`);
-  console.log();
+  log(`Installing dependencies using ${command}...`);
+  log(); // why is this here
```
because it's ugly otherwise and i apparently forgot how newlines work
